### PR TITLE
Bring compatibility with Channels v3

### PR DESCRIPTION
**Pull request by fabienheureux**
Channels v3 is currently not supported.
This bring changes from #74 alongside an import that does not work with channels because it was renamed in https://github.com/django/channels/commit/4f320745278ab45ac52f9d622dda51f2f45bffc8#diff-8adaf2e6e01ec2cc05dd10bfd212d6dddfadabe0ddc679b821e03024806647b7

Also in this PR, I added github actions in order to have the tests running again on push/PR https://github.com/fabienheureux/graphql-ws/runs/3380650273?check_suite_focus=true

### CODE REVIEW

**Review by jkimbo — APPROVED** [review]
Changes look fine but it looks like tests aren't running. Probably worth moving to github actions?

**Comment by fabienheureux** [comment]
> 
> 
> Changes look fine but it looks like tests aren't running. Probably worth moving to github actions?

I added a workflow/ci.yml file, but I don't understand why actions are not ran.
Any idea ? Is there something needed to enable Github actions in the repo's settings maybe ?

**Comment by fabienheureux** [comment]
Ha nevermind, they are ran on the fork https://github.com/fabienheureux/graphql-ws/actions

**Comment by fabienheureux** [comment]
@jkimbo tests are now running, and they pass 👍  https://github.com/fabienheureux/graphql-ws/runs/3380650273?check_suite_focus=true

File: Makefile [hunk 1]
```diff
@@ -50,9 +50,12 @@ clean-test: ## remove test and coverage artifacts
 lint: ## check style with flake8
 	flake8 graphql_ws tests
 
+format:
+	black ./graphql_ws ./tests
```
Comment (fabienheureux) [inline Makefile:54]: I added a script to format the code locally. 
It helps solve flake8 reported issues. Let me know if this is out of the scope of this PR

File: graphql_ws/base.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def remove_operation(self, op_id):
 
     def unsubscribe(self, op_id):
         async_iterator = self.remove_operation(op_id)
-        if hasattr(async_iterator, 'dispose'):
+        if hasattr(async_iterator, "dispose"):
```
Comment (fabienheureux) [inline graphql_ws/base.py:44]: I ran black on the repo locally to fix various code style issues

File: setup.cfg [hunk 3]
```diff
@@ -50,13 +51,12 @@ test =
...
     gevent
-    graphene>=2.0
     graphene_django
     mock; python_version<"3"
     django==1.11.*; python_version<"3"
     channels==1.*; python_version<"3"
-    django==2.*; python_version>="3"
-    channels==2.*; python_version>="3"
+    django==3.*; python_version>="3"
+    channels==3.*; python_version>="3"
```
Comment (fabienheureux) [inline setup.cfg:59]: Is it safe to assume all python >3 users will use django ?

File: tests/test_base.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ def test_observer_data():
         send_error=send_error,
         send_message=send_message,
     )
-    observer.on_next('data')
+    observer.on_next("data")
```
Comment (fabienheureux) [inline tests/test_base.py:94]: I ran black on the repo locally to fix various code style issues

File: tests/test_base.py [hunk 5]
```diff
@@ -107,6 +107,6 @@ def test_observer_exception():
         send_error=send_error,
         send_message=send_message,
     )
-    observer.on_next(TypeError('some bad message'))
+    observer.on_next(TypeError("some bad message"))
```
Comment (fabienheureux) [inline tests/test_base.py:110]: ditto

File: tests/test_base_async.py [hunk 6]
```diff
@@ -83,18 +83,18 @@ async def test_resolver_with_promise(server):
         connection_context=None, op_id=1, execution_result=result
     )
     assert server.send_message.called
-    assert result.data == {'test': [1, 2]}
+    assert result.data == {"test": [1, 2]}
```
Comment (fabienheureux) [inline tests/test_base_async.py:86]: ditto

File: Makefile (hunk 1 above)
Comment (jkimbo) [reply Makefile:54]: Makes sense to me 👍

File: setup.cfg (hunk 3 above)
Comment (jkimbo) [reply setup.cfg:59]: Not sure what you mean?

Comment (fabienheureux) [reply setup.cfg:59]: I meant "will use django 3"

**Pull request merged**